CHANGELOG.md

```diff
@@ -0,0 +1,72 @@
+# V2.0.0
+
+## New features
+Most new features are centered around better typings
+
+### EventDispatcher can be restricted to certain event classes
+You can now pass a generic parameter to `EventDispatcher` indicating
+which type of events it will dispatch. Besides it being documentation to
+other developers about which events to expect, it also allows this library
+to add more useful typings to methods like `addEventListener`. Example:
+
+```ts
+class Scheduler extends EventDispatcher<SchedulerEvent> {
+```
+
+### It is now easier to create event classes
+Event classes can still be created by manually extending `AbstractEvent`.
+However, this is somewhat verbose. Moreover, it is very common to want to
+have some kind of `data` property on an event. This is now possible using
+2 new utilities: `createEventClass` and `createIsomorphicEventClass`. See
+the [readme](README.md) for more info.
+
+### Event classes can have isomorphic data or options
+Previously an event class was either fixed to a set of options and a certain
+type of `data`, or it was type too loosely. It is now possible to have the
+options and the `data` depend on the event `type`. 
+
+## Changes
+
+### Static string types on Event classes
```

Comment:
Reason for this? Only to allow typing?

Reply:
It was more convenient to write typings for. It is _maybe_ possible to do it the old way, but I thought this new way actually looks quite clean. What would you prefer (if typing wasn't an issue) ?

Reply:
I guess it makes sense, it's just different than what people might expect. Although having typed events is not common at all in the DOM :) No hard opinion, keep it like this!